### PR TITLE
Py decimal proto

### PR DESCRIPTION
**Pull request by fm-andres-rangel**

### CODE REVIEW

File: include/libmpdec/mpdecimal.h [hunk 1]
```diff
@@ -0,0 +1,946 @@
...
+int mpd_resize(mpd_t *result, mpd_ssize_t nwords, mpd_context_t *ctx);
+int mpd_resize_zero(mpd_t *result, mpd_ssize_t nwords, mpd_context_t *ctx);
+
+MPD_PRAGMA(MPD_HIDE_SYMBOLS_END) /* restore previous scope rules */
+
+#ifdef __cplusplus
+} /* END extern "C" */
+#endif
+
+#endif /* LIBMPDEC_MPDECIMAL_H_ */
```
Comment (fm-ivang) [inline include/libmpdec/mpdecimal.h:946]: newline

File: python/include/extractor/python/decimal128.hpp [hunk 2]
```diff
@@ -618,3 +621,35 @@ fmc_decimal128_t Decimal128_val(PyObject *obj) {
 PyObject *Decimal128_new(fmc_decimal128_t obj) {
   return ExtractorBaseTypeDecimal128::py_new(obj);
 }
+
+PyObject *ExtractorBaseTypeDecimal128::as_decimal(PyObject *self,
+                                                  PyObject *args) {
+  static PyObject *dectype = PyDecimal_Type();
```
Comment (fm-ivang) [inline python/include/extractor/python/decimal128.hpp:627]: I think it does not need to be static, as PyDecimal_Type() already returns a static ptr

**Review by fm-ivang — CHANGES_REQUESTED** [review]

File: .github/workflows/cmake-pr.yml [hunk 3]
```diff
@@ -64,7 +69,10 @@ jobs:
         submodules: recursive
 
     - name: Install dependencies
-      run: brew install pkg-config cmake googletest python
+      run: |
+        brew install pkg-config cmake googletest python mpdecimal
+        brew link mpdecimal
```
Comment (fm-ivang) [inline .github/workflows/cmake-pr.yml:74]: Remove brew link mpdecimal

File: python/CMakeLists.txt [hunk 4]
```diff
@@ -40,6 +41,7 @@ target_compile_definitions(
     PRIVATE
     PY_EXTR_VER="${PROJECT_VERSION}"
     NO_DLL_DECORATOR
+    UNIVERSAL
```
Comment (fm-ivang) [inline python/CMakeLists.txt:44]: review

**Review by fm-ivang — CHANGES_REQUESTED** [review]

**Review by fm-ivang — DISMISSED** [review]

**Review by fm-ivang — APPROVED** [review]

**Review by fm-fede-ravchina — APPROVED** [review]

**Pull request merged**